Cleanup and compiler issues only, no functional change

Changes partly very old and not possible to find the original author.

Signed-off-by: Thomas Renninger <trenn@suse.de>

Index: ipmitool-1.8.18/lib/ipmi_ekanalyzer.c
===================================================================
--- ipmitool-1.8.18.orig/lib/ipmi_ekanalyzer.c	2018-09-09 13:48:58.000000000 +0200
+++ ipmitool-1.8.18/lib/ipmi_ekanalyzer.c	2018-09-25 11:42:30.901490572 +0200
@@ -4014,7 +4014,7 @@ ipmi_ek_display_clock_config_record(stru
 					(feature & 1) ? "Source" : "Receiver");
 			printf("\tFamily:  0x%02x    - AccLVL: 0x%02x\n",
 					family, accuracy);
-			printf("\tFRQ: %-9ld - min: %-9ld - max: %-9ld\n",
+			printf("\tFRQ: %-9lu - min: %-9lu - max: %-9lu\n",
 					freq, min_freq, max_freq);
 		}
 		printf("\n");
Index: ipmitool-1.8.18/lib/ipmi_picmg.c
===================================================================
--- ipmitool-1.8.18.orig/lib/ipmi_picmg.c	2018-09-09 13:48:58.000000000 +0200
+++ ipmitool-1.8.18/lib/ipmi_picmg.c	2018-09-25 11:42:30.905490809 +0200
@@ -850,7 +850,7 @@ ipmi_picmg_portstate_get(struct ipmi_int
 					}
 					else if (d->type >= 0x06 && d->type <= 0xef)
 					{
-						printf("Reserved\n");
+						printf("Reserved %d\n",d->type);
 					}
 					else if (d->type >= 0xf0 && d->type <= 0xfe)
 					{
@@ -1702,7 +1702,7 @@ ipmi_picmg_clk_get(struct ipmi_intf * in
 						oemval2str( rsp->data[3], rsp->data[4],
 											picmg_clk_accuracy_vals));
 		
-		      printf("  - Freq:   %ld\n", freq);
+		      printf("  - Freq:   %lu\n", freq);
 		   }
 		}
 	}
Index: ipmitool-1.8.18/lib/ipmi_sdr.c
===================================================================
--- ipmitool-1.8.18.orig/lib/ipmi_sdr.c	2018-09-09 13:48:58.000000000 +0200
+++ ipmitool-1.8.18/lib/ipmi_sdr.c	2018-09-25 11:42:30.909491047 +0200
@@ -52,6 +52,7 @@
 #include <ipmitool/ipmi_entity.h>
 #include <ipmitool/ipmi_constants.h>
 #include <ipmitool/ipmi_strings.h>
+#include <ipmitool/ipmi_sensor.h>
 
 #if HAVE_CONFIG_H
 # include <config.h>
